server/controllers/match/bedSearchController.ts

```diff
@@ -1,41 +1,37 @@
 import type { Request, RequestHandler, Response } from 'express'
+
 import { BedSearchParametersUi } from '../../@types/ui'
+import paths from '../../paths/match'
 import { PersonService } from '../../services'
 import BedService from '../../services/bedService'
 
+import { startDateFromParams } from '../../utils/matchUtils'
+
 export default class BedSearchController {
   constructor(private readonly bedService: BedService, private readonly personService: PersonService) {}
 
   search(): RequestHandler {
     return async (req: Request, res: Response) => {
       const query = Object.keys(req.query).length ? req.query : undefined
+      const body = Object.keys(req.body).length ? req.body : undefined
+
+      if (query || body) {
+        const params = {
+          ...query,
+          ...body,
+        }
```

Comment:
Could we put this call above the conditional and then have `if (params) {` in the conditional? There might be some weird TS/JS reason why not, so it's no biggie if you'va already tried it / it doesn't work.

Reply:
🙌